src/lib/Cashlink.ts

```diff
@@ -0,0 +1,415 @@
+                if (!balance && !pendingFundingTx) {
+                    // Handle expired/replaced funding tx
+                    newState = CashlinkState.UNCHARGED;
+                    // Not break;ing here, because we need to see if the cashlink is already CLAIMED.
+                }
+                if (knownFundingTx) {
+                    newState = CashlinkState.UNCLAIMED;
+                } else break; // If no known transactions are found, no further checks are necessary
+            case CashlinkState.UNCLAIMED:
+                if (pendingClaimingTx) {
+                    newState = CashlinkState.CLAIMING;
+                }
+            case CashlinkState.CLAIMING:
+                if (balance) {
+                    // Handle recharged/reused cashlink
+                    if (!pendingClaimingTx) newState = CashlinkState.UNCLAIMED;
+                    break; // If a balance is detected on the cashlink, it cannot be in CLAIMED state.
+                }
+                if (knownClaimingTx) {
+                    newState = CashlinkState.CLAIMED;
+                }
+            case CashlinkState.CLAIMED:
+                // Detect cashlink re-use and chain rebranches
+                if (pendingFundingTx) newState = CashlinkState.CHARGING;
+                if (balance) newState = CashlinkState.UNCLAIMED;
+                if (pendingClaimingTx) newState = CashlinkState.CLAIMING;
+        }
+
+        if (newState !== this.state) this._updateState(newState);
+    }
+
+    public toObject(): CashlinkEntry {
+        return {
+            keyPair: new Uint8Array(this.keyPair.serialize()),
+            address: this.address.toUserFriendlyAddress(),
+            value: this.value,
+            message: this.message,
+            state: this.state,
+            timestamp: this.timestamp,
+            contactName: this.contactName,
+        };
+    }
+
+    public render() {
+        const buf = new Nimiq.SerialBuffer(
+            /*key*/ this.keyPair.privateKey.serializedSize +
+            /*value*/ 8 +
+            /*message length*/ (this._messageBytes.byteLength ? 1 : 0) +
+            /*message*/ this._messageBytes.byteLength,
+        );
+
+        this.keyPair.privateKey.serialize(buf);
+        buf.writeUint64(this.value);
+        if (this._messageBytes.byteLength) {
+            buf.writeUint8(this._messageBytes.byteLength);
+            buf.write(this._messageBytes);
+        }
+
+        let result = Nimiq.BufferUtils.toBase64Url(buf);
+        // replace trailing . by = because of URL parsing issues on iPhone.
+        result = result.replace(/\./g, '=');
+        // iPhone also has a problem to parse long words with more then 300 chars in a URL in WhatsApp
+        // (and possibly others). Therefore we break the words by adding a ~ every 256 characters in long words.
+        result = result.replace(/[A-Za-z0-9_]{257,}/g, (match) => match.replace(/.{256}/g, '$&~'));
+
+        return result;
```

Comment:
```
        return Nimiq.BufferUtils.toBase64Url(buf)
        // replace trailing . by = because of URL parsing issues on iPhone.
            .replace(/\./g, '=')
        // iPhone also has a problem to parse long words with more then 300 chars in a URL in WhatsApp
        // (and possibly others). Therefore we break the words by adding a ~ every 256 characters in long words.
            .replace(/[A-Za-z0-9_]{257,}/g, (match) => match.replace(/.{256}/g, '$&~'));
```
You could just do that, right ? No real need for a `result` variable.

Reply:
Yes. I think this was done here for readability of the code. So the different steps can be explained by itself.
@danimoh?

Reply:
Yes, i think it reads nicer like this.